### PR TITLE
Resolves #1374 - Provide example of using context

### LINKED ISSUE #1374: Preview provides empty context to rendered components
**Current behavior**

Preview renders by reactDom.render method, which not pass root react context to rendered components

https://github.com/styleguidist/react-styleguidist/blob/master/src/client/rsg-components/Preview/Preview.js#L79

**Expected behavior**

Rendered components should have the same react context as demo.

In my case I faced with this problem when switched demo theme. All except rendered components gets new theme

### PR DESCRIPTION
**Pull request by jkarttunen**
It's not perfect, but it is a start.

### CODE REVIEW

**Comment by jkarttunen** [comment]
Is this ok? The default button style seems to be pink dotted line. 

Screenshot 2019-07-29 at 16 24 41

**Comment by sapegin** [comment]
Yup, it's here to catch isolation bugs ;-)

**Comment by jkarttunen** [comment]
Maybe i should add custom styling to that props panel then?

**Comment by sapegin** [comment]
I don't know ;-) Ideally a button should have all the necessary styles.

**Comment by jkarttunen** [comment]
Ok, consider the change request done then. (Not sure if there is a button for me to do this)

File: examples/sections/src/components/ThemeButton/Readme.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+An example about using react context with a component
```
Comment (sapegin) [inline examples/sections/src/components/ThemeButton/Readme.md]:
```suggestion
An example of using React Context with a component:
```

File: docs/Documenting.md [hunk 2]
```diff
@@ -310,6 +310,8 @@ class ModalExample extends React.Component {
 ;<ModalExample />
 ````
 
+If component consumes React context, you need to have a context provider in the example or in a custom Wrapper. See [ThemeButton example](https://github.com/styleguidist/react-styleguidist/tree/master/examples/basic/src/components/ThemeButton).
```
Comment (sapegin) [inline docs/Documenting.md]:
```suggestion
If a component consumes React Context, you need a context provider in the example or in a custom `Wrapper` component. See [ThemeButton example](https://github.com/styleguidist/react-styleguidist/tree/master/examples/sections/src/components/ThemeButton).
```

File: examples/sections/src/components/ThemeButton/ThemeButton.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.theme-button--light,
```
Comment (sapegin) [inline examples/sections/src/components/ThemeButton/ThemeButton.css]: Let's use a bit saner naming convention here:

```css
.ThemeButton { /* shared styles */ }
.ThemeButton--dark {}
.ThemeButton--light {}
```

Comment (jkarttunen) [reply examples/sections/src/components/ThemeButton/ThemeButton.css]: Sure

File: examples/sections/src/ThemeContext.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+
+/**
+ * Button that counts how many times it was pressed and exposes a `@public` method to reset itself.
```
Comment (sapegin) [inline examples/sections/src/ThemeContext.js]: This comment looks out of place.

Comment (jkarttunen) [reply examples/sections/src/ThemeContext.js]: Fixed

**Comment by sapegin** [comment]
Thanks, merging! 🌞

**Pull request merged**